enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/config/ConfigHolder.java

```diff
@@ -228,6 +229,8 @@ private void populateAuthService(Service cdsAuth) {
         authDto.setMaxMessageSize(cdsAuth.getMaxMessageSize());
 
         ThreadPoolConfig threadPool = new ThreadPoolConfig();
+        MBeanRegistrator.registerMBean(threadPool);
```

Comment:
Do we need to registerMBean only if the JMX Enabled condition is satisfied? If there is no impact it is kinda okay to even keep IMO.

Reply:
Since we have a added system property to check whether jmx enabled, I added that check inside the `registerMBean` and will be safe for all mbean registrations.